### PR TITLE
[dagster-airlift] bubble up translation fn

### PR DESCRIPTION
**Pull request by dpeng817**
## Changelog
NOCHANGELOG

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24891?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24892** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24892?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24891** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24891?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24831** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24831?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#24832](https://github.com/dagster-io/dagster/pull/24832) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24832?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#24830** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24830?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24829** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24829?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24828** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24828?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by schrockn — COMMENTED** [review]
Got it so the implication is that we are going to make `DagRun`, `TaskInstance`, and `AirflowDefinitionData`, as well as function to make a "default" materialization public, correct?

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor/event_translation.py [hunk 1]
```diff
@@ -13,6 +13,10 @@
 from dagster_airlift.core.airflow_defs_data import AirflowDefinitionsData
 from dagster_airlift.core.airflow_instance import DagRun, TaskInstance
 
+AirflowEventTranslationFn = Callable[
+    [DagRun, Sequence[TaskInstance], AirflowDefinitionsData], List[AssetMaterialization]
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/core/sensor/event_translation.py:17]: Might as well be Iterable[AssetMaterialization] so people can build a function that yields

**Review by schrockn — CHANGES_REQUESTED** [review]
Just want to make sure we understand the implications here.

**Comment by dpeng817** [comment]
@schrockn that's mostly correct: I think there's some workshopping to be done on the final API. Where I'd like to end up is here:
`Callable[DagRun, Sequence[TaskInstance]], Iterable[AssetEvent]]` where AssetEvent is a union of materialization, observation, and check eval result.

File: examples/experimental/dagster-airlift/dagster_airlift/core/sensor/event_translation.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/sensor/event_translation.py:17]: works for me

**Review by schrockn — APPROVED** [review]

**Pull request merged**